typescript/infra/src/config/agent/relayer.ts

```diff
@@ -45,17 +49,31 @@ export interface BaseRelayerConfig {
   addressBlacklist?: string;
   transactionGasLimit?: BigNumberish;
   skipTransactionGasLimitFor?: string[];
-  metricAppContexts?: MetricAppContext[];
+  metricAppContextsGetter?: () => MetricAppContext[];
 }
 
 // Full relayer-specific agent config for a single chain
 export type RelayerConfig = Omit<RelayerAgentConfig, keyof AgentConfig>;
+// Config intended to be set as configMap values, these are usually really long
+// and are intended to derisk hitting max env var length limits.
+export type RelayerConfigMapConfig = Pick<
+  RelayerConfig,
+  'addressBlacklist' | 'gasPaymentEnforcement' | 'metricAppContexts'
+>;
+// The rest of the config is intended to be set as env vars.
```

Comment:
do we enforce this? might be nice to use zod on process.env

Reply:
not enforced, this is just a practice I'm having us do internally. Because these are env vars piped into the agent image i think zod will be hard / overkill